compiler/resolution/preFold.cpp

```diff
@@ -2334,6 +2334,12 @@ static Expr* resolveTupleIndexing(CallExpr* call, Symbol* baseVar) {
       USR_FATAL_CONT(call, "tuple index %" PRIu64 " is out of bounds", uindex);
       error = true;
     }
+  } else if (get_bool(call->get(3), &uindex)) {
+    sprintf(field, "x%" PRIu64, uindex);
+    if (uindex >= (unsigned long)baseType->fields.length-1) {
+      USR_FATAL_CONT(call, "tuple index %" PRIu64 " is out of bounds", uindex);
+      error = true;
+    }
```

Comment:
Should this case be INT_FATAL? It looks like we can't have 0-sized tuples today, so having this fail can only mean we have messed up something in the compiler.

Reply:
We do have 1-tuples, though...  (and hope to have 0-tuples)

Reply:
Oops, right. I don't know what I was thinking.

On a similar vein, though, is it concerning to be able to bool-index a tuple that has more than 2 items?

Reply:
I don't think so.  It seems as reasonable as it is to use an `int(8)` value to index into an array that has > 256 items.  It won't get you access to everything, but that doesn't make it unreasonable to do.

Reply:
That's fair.